src/services/GroupService/index.ts

```diff
@@ -0,0 +1,54 @@
+import {
+    Service,
+} from "typedi";
+import {
+    Permission,
+} from "./types";
+import GroupModel from "../../models/GroupModel";
+import UserModel from "../../models/UserModel";
+import {getManager} from "typeorm";
+
+@Service()
+export default class GroupService {
+    async getById(id: string) {
+        return GroupModel.findOneOrFail(id);
+    }
+
+    async createGroup(name, permissions: Permission[]) {
+        const group = await GroupModel.save(
+            GroupModel.create({
+                name,
+                permissions,
+            }),
+        );
+
+        return group.id;
+    }
+
+    async updateGroup(id: string, name: string, permissions: Permission[]) {
+            const group = await this.getById(id);
+            if (name) {
+                group.name = name;
+            }
```

Comment:
mutation of an object is not really functional approache. It's better to avoid it

Reply:
Done.